src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -557,20 +570,80 @@ def delete_ag_url_path_map_rule(resource_group_name, application_gateway_name, u
         resource_group_name, application_gateway_name, ag)
 
 def set_ag_waf_config(resource_group_name, application_gateway_name, enabled,
-                      firewall_mode=ApplicationGatewayFirewallMode.detection.value, no_wait=False):
+                      firewall_mode=ApplicationGatewayFirewallMode.detection.value,
+                      rule_set_type='OWASP', rule_set_version=None, disabled_rule_groups=None,
+                      disabled_rules=None, no_wait=False):
     ApplicationGatewayWebApplicationFirewallConfiguration = get_sdk(
         ResourceType.MGMT_NETWORK,
         'ApplicationGatewayWebApplicationFirewallConfiguration', mod='models')
     ncf = _network_client_factory().application_gateways
     ag = ncf.get(resource_group_name, application_gateway_name)
-    ag.web_application_firewall_configuration = \
-        ApplicationGatewayWebApplicationFirewallConfiguration(enabled == 'true', firewall_mode)
+    if supported_api_version(ResourceType.MGMT_NETWORK, min_api='2017-03-01'):  # pylint: disable=too-many-nested-blocks
+        ag.web_application_firewall_configuration = \
+            ApplicationGatewayWebApplicationFirewallConfiguration(
+                enabled == 'true', firewall_mode, rule_set_type, rule_set_version)
+        if disabled_rule_groups or disabled_rules:
+            ApplicationGatewayFirewallDisabledRuleGroup = get_sdk(
+                ResourceType.MGMT_NETWORK,
+                'ApplicationGatewayFirewallDisabledRuleGroup', mod='models')
+
+            disabled_groups = []
+
+            # disabled groups can be added directly
+            for group in disabled_rule_groups or []:
+                disabled_groups.append(ApplicationGatewayFirewallDisabledRuleGroup(group))
+
+            # for disabled rules, we have to look up the IDs
+            if disabled_rules:
+                results = list_ag_waf_rule_sets(
+                    ncf, _type=rule_set_type, version=rule_set_version, group='*')
+                for item in results:
+                    for group in item.rule_groups:
+                        disabled_group = ApplicationGatewayFirewallDisabledRuleGroup(
+                            group.rule_group_name, [])
+
+                        for rule in group.rules:
+                            if str(rule.rule_id) in disabled_rules:
+                                disabled_group.rules.append(rule.rule_id)
+                        if disabled_group.rules:
+                            disabled_groups.append(disabled_group)
+            ag.web_application_firewall_configuration.disabled_rule_groups = disabled_groups
+    else:
+        ag.web_application_firewall_configuration = \
+            ApplicationGatewayWebApplicationFirewallConfiguration(
+                enabled == 'true', firewall_mode)
```

Comment:
In this case, I would think that just having two different methods then having one if statement in commands.py (like storage account create) would be simpler.
It would also mean that you don't need the ignore_types in params.